bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x/src/main/java/com/google/cloud/bigtable/mirroring/hbase1_x/utils/mirroringmetrics/MirroringSpanFactory.java

```diff
@@ -0,0 +1,285 @@
+  }
+
+  public <T> FutureCallback<T> wrapReadVerificationCallback(final FutureCallback<T> callback) {
+    return new FutureCallback<T>() {
+      @Override
+      public void onSuccess(@NullableDecl T t) {
+        try (Scope scope = MirroringSpanFactory.this.verificationScope()) {
+          callback.onSuccess(t);
+        }
+      }
+
+      @Override
+      public void onFailure(Throwable throwable) {
+        try (Scope scope = MirroringSpanFactory.this.verificationScope()) {
+          callback.onFailure(throwable);
+        }
+      }
+    };
+  }
+
+  public <T> WriteOperationFutureCallback<T> wrapWriteOperationCallback(
+      final WriteOperationFutureCallback<T> callback) {
+    return new WriteOperationFutureCallback<T>() {
+      @Override
+      public void onFailure(Throwable throwable) {
+        try (Scope scope = MirroringSpanFactory.this.writeErrorScope()) {
+          callback.onFailure(throwable);
+        }
+      }
+    };
+  }
+
+  public Scope flowControlScope() {
+    return flowControlSpanBuilder().startScopedSpan();
+  }
+
+  public Scope verificationScope() {
+    return tracer.spanBuilder("verification").startScopedSpan();
+  }
+
+  public Scope writeErrorScope() {
+    return tracer.spanBuilder("writeErrors").startScopedSpan();
+  }
+
+  public Scope operationScope(HBaseOperation name) {
+    return new MirroringOperationScope(name);
+  }
+
+  public Span getCurrentSpan() {
+    return tracer.getCurrentSpan();
+  }
+
+  public Scope scheduleFlushScope() {
+    return tracer.spanBuilder("scheduleFlush").startScopedSpan();
+  }
+
+  public Scope spanAsScope(Span span) {
+    return tracer.withSpan(span);
+  }
+
+  private <T> T wrapPrimaryOperationAndMeasure(
+      CallableThrowingIOAndInterruptedException<T> operationRunner, HBaseOperation operationName)
+      throws IOException, InterruptedException {
+    return wrapOperationAndMeasure(
+        operationRunner,
+        PRIMARY_LATENCY,
+        PRIMARY_ERRORS,
+        this.primaryOperationScope(),
+        operationName);
+  }
+
+  private <T> T wrapSecondaryOperationAndMeasure(
+      CallableThrowingIOAndInterruptedException<T> operationRunner, HBaseOperation operationName)
+      throws IOException, InterruptedException {
+    return wrapOperationAndMeasure(
+        operationRunner,
+        SECONDARY_LATENCY,
+        SECONDARY_ERRORS,
+        this.secondaryOperationsScope(),
+        operationName);
+  }
+
+  private <T> T wrapOperationAndMeasure(
+      CallableThrowingIOAndInterruptedException<T> operationRunner,
+      MeasureLong latencyMeasure,
+      MeasureLong errorMeasure,
+      Scope scope,
+      HBaseOperation operationName)
+      throws IOException, InterruptedException {
+    boolean operationFailed = false;
+
+    Stopwatch stopwatch = Stopwatch.createUnstarted();
+    try (Scope scope1 = scope) {
+      stopwatch.start();
+      return operationRunner.call();
+    } catch (IOException | InterruptedException e) {
```

Comment:
Why not catch Exception or Throwable? if any other RuntimeException is thrown, you will not log it as an error?